### PR TITLE
Better `ci.pipeline.run.cause` attribute

### PR DESCRIPTION
**Pull request by cyrille-leclerc**
Better `ci.pipeline.run.cause` attribute, finish the work started on 
- https://github.com/jenkinsci/opentelemetry-plugin/pull/187  .

Replace `ci.pipeline.run.cause.description` and `ci.pipeline.run.cause.xxx` to just keep `ci.pipeline.run.cause`with a machine readable value like `UserIdCause:anonymous` or `BranchIndexingCause `

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by cyrille-leclerc** [comment]
@v1v I finished the work started on 
- https://github.com/jenkinsci/opentelemetry-plugin/pull/187

Can you please review?

**Comment by cyrille-leclerc** [comment]
Merge so I can move forward on 
- https://github.com/jenkinsci/opentelemetry-plugin/issues/204
- https://github.com/jenkinsci/opentelemetry-plugin/discussions/98

File: src/test/java/io/jenkins/plugins/opentelemetry/JenkinsOtelPluginMBPIntegrationTest.java [hunk 1]
```diff
@@ -74,8 +75,7 @@ public void testMultibranchPipelineStep() throws Exception {
         MatcherAssert.assertThat(attributes.get(JenkinsOtelSemanticAttributes.CI_PIPELINE_MULTIBRANCH_TYPE), CoreMatchers.is(OtelUtils.BRANCH));
         MatcherAssert.assertThat(attributes.get(JenkinsOtelSemanticAttributes.CI_PIPELINE_TYPE), CoreMatchers.is(OtelUtils.MULTIBRANCH));
         MatcherAssert.assertThat(attributes.get(JenkinsOtelSemanticAttributes.CI_PIPELINE_RUN_DESCRIPTION), CoreMatchers.is("Bar"));
-        MatcherAssert.assertThat(attributes.get(JenkinsOtelSemanticAttributes.CI_PIPELINE_RUN_CAUSE_XXX), CoreMatchers.is(CoreMatchers.notNullValue()));
-        MatcherAssert.assertThat(attributes.get(JenkinsOtelSemanticAttributes.CI_PIPELINE_RUN_CAUSE_DESCRIPTION), CoreMatchers.is(CoreMatchers.notNullValue()));
+        MatcherAssert.assertThat(attributes.get(JenkinsOtelSemanticAttributes.CI_PIPELINE_RUN_CAUSE), CoreMatchers.is(Arrays.asList("BranchIndexingCause")));
```
Comment (v1v) [inline src/test/java/io/jenkins/plugins/opentelemetry/JenkinsOtelPluginMBPIntegrationTest.java:78]: 💯

**Review by v1v — COMMENTED** [review]
💯

**Pull request merged**